src/goto-instrument/contracts/contracts.cpp

```diff
@@ -274,7 +282,8 @@ void code_contractst::check_apply_loop_contracts(
           log.debug() << ", ";
         ran_once = true;
         log.debug() << format(target);
-        loop_assigns.add_to_write_set(target);
+        auto payload = instrument_spec_assigns.track_spec_target(target);
+        snapshot_instructions.destructive_append(payload);
```

Comment:
Would it perhaps make for a better interface if `snapshot_instructions` were passed as an argument?